shared/util/saga.js

```diff
@@ -125,25 +125,62 @@ function* sequentially(effects: Array<any>): Generator<any, Array<any>, any> {
 // whatever purework returns will be yielded on.
 // i.e. it can return put(someAction). That effectively transforms the input action into another action
 // It can also return all([put(action1), put(action2)]) to dispatch multiple actions
-function safeTakeEveryPure<A, R, FinalAction>(
+function safeTakeEveryPure<A, R, FinalAction, FinalActionError>(
   pattern: string | Array<any> | Function,
   pureWorker: ((action: A, state: TypedState) => any) | ((action: A) => any),
-  actionCreatorsWithResult?: (result: R) => FinalAction
+  actionCreatorsWithResult?: (result: R) => FinalAction,
+  actionCreatorsWithError?: (result: R) => FinalActionError
 ) {
   return safeTakeEvery(pattern, function* safeTakeEveryPureWorker(action: A) {
     // If the pureWorker fn takes two arguments, let's pass the state
-    let result
-    if (pureWorker.length === 2) {
-      const state: TypedState = yield select()
-      // $FlowIssue - doesn't understand checking for arity
-      result = yield pureWorker(action, state)
-    } else {
-      // $FlowIssue - doesn't understand checking for arity
-      result = yield pureWorker(action)
+    try {
+      let result
+      if (pureWorker.length === 2) {
+        const state: TypedState = yield select()
+        // $FlowIssue - doesn't understand checking for arity
+        result = yield pureWorker(action, state)
+      } else {
+        // $FlowIssue - doesn't understand checking for arity
+        result = yield pureWorker(action)
+      }
+
+      if (actionCreatorsWithResult) {
+        yield actionCreatorsWithResult(result)
+      }
+    } catch (e) {
+      if (actionCreatorsWithError) {
+        yield actionCreatorsWithError(e)
+      }
     }
+  })
+}
+// Similar to safeTakeEveryPure
+function safeTakeLatestPure<A, R, FinalAction, FinalActionError>(
```

Comment:
a take latest version

Reply:
I don't think that makes sense. If this is a pure function there is no delay, it just runs.

Reply:
The framework itself has a yield in it so it could technically cancel. If you pass all or other things (sequentially) that also could cancel with a new incoming action right?

Reply:
What happens when the call is cancelled? Does the error case get called then? If so I think it's okay

Reply:
it should, you get the throw and you can do a yield cancelled to figure that out, in practice we don't handle the cancel cases at this level that i've seen